checkov/common/util/secrets.py

```diff
@@ -183,13 +195,21 @@ def omit_secret_value_from_graph_checks(check: BaseGraphCheck, check_result: dic
         }
 
     if resource_attributes_to_omit:
+        # Universal mask ('*') might exist in resource_attributes_to_omit. If it does exist, we need to mask the all
+        # entities in resource types according to resource_attributes_to_omit.get('*')
+        universal_mask = set(resource_attributes_to_omit.get(RESOURCE_ATTRIBUTES_TO_OMIT_UNIVERSAL_MASK, set()))
         for resource in check.resource_types:
-            if resource in resource_attributes_to_omit:
-                for attribute in resource_attributes_to_omit.get(resource):  # type:ignore[union-attr]
-                    if attribute in entity_config:
-                        secret = entity_config.get(attribute)
-                        if isinstance(secret, list) and secret:
-                            secrets.add(secret[0])
+            resource_masks = set(resource_attributes_to_omit.get(resource, set()))
+            # resource_masks should contain all mask rules that should apply on this resource
+            resource_masks.update(universal_mask)
```

Comment:
Variable `universal_mask` saving and the update function can be skipped here as well for most scans if we first check if univeral mask exists like in the previous suggestion

Reply:
same